find-unused-vars.coffee

```diff
@@ -0,0 +1,179 @@
+esprima    = require 'esprima'
+nodeType   = require './node-type'
+parseRegex = require './parse-regex'
+
+
+processResults = (results, path) ->
+  for name of results
+    if results.hasOwnProperty name
+      stats = results[name]
+      if stats.declarations is 0 and stats.calls is 0
+        console.log 'Variable', name, 'undeclared', ' in ', path
+      else if stats.declarations > 1 and stats.calls is 0
+        console.log 'Variable', name, 'declared multiple times', ' in ', path
+      else if stats.calls is 0
+        console.log 'Variable', name, 'declared but not called', ' in ', path
+
+
+traverse = (node, func) ->
+  func node
+  for key of node
+    if node.hasOwnProperty key
+      child = node[key]
+      if typeof child is 'object' and child isnt null
+        if Array.isArray child
+          child.forEach (node) ->
+            traverse node, func
+        else
+          traverse child, func
+
+
+checkIdentifier = (node) -> node.type is nodeType.Identifier
```

Comment:
Do we really need this as a separate function? Other comparisons have not been done like this, they are simple checks, but this one is wrapped in a function. Why?